core/src/main/java/org/mskcc/cbio/oncokb/util/IndicatorUtils.java

```diff
@@ -234,17 +255,19 @@ public static IndicatorQueryResp processQuery(Query query, String geneStatus,
                 Alteration oncogenicMutation = AlterationUtils.findAlteration(gene, "Oncogenic Mutations");
                 if (oncogenicMutation != null) {
                     relevantAlterations.add(oncogenicMutation);
-                    treatmentEvidences.addAll(EvidenceUtils.keepHighestLevelForSameTreatments(
-                        EvidenceUtils.convertEvidenceLevel(
-                            EvidenceUtils.getEvidence(Collections.singletonList(oncogenicMutation),
-                                MainUtils.getTreatmentEvidenceTypes(),
-                                (levels != null ?
-                                    new HashSet<>(CollectionUtils.intersection(levels,
-                                        LevelUtils.getPublicAndOtherIndicationLevels())) : LevelUtils.getPublicAndOtherIndicationLevels())), new HashSet<>(oncoTreeTypes))));
+                    if (hasTreatmentEvidence) {
+                        treatmentEvidences.addAll(EvidenceUtils.keepHighestLevelForSameTreatments(
+                            EvidenceUtils.convertEvidenceLevel(
+                                EvidenceUtils.getEvidence(Collections.singletonList(oncogenicMutation),
```

Comment:
Same questions here. Should we do the null check before passing the return as the parameter for the next round call?

Reply:
same here, an empty list will always be returned.